src/Tutor.Web/Extensions/ResultToResponseExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿using FluentResults;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+
+namespace Tutor.Web.Extensions
+{
+    public static class ResultToResponseExtensions
+    {
+        // return result.CreateResponse(Ok, CreateErrorResponse);
+        public static ActionResult CreateResponse (
+            this Result result, 
+            Func<ActionResult> onSuccess, Func<List<IError>, ActionResult> onFailure)
+        {
+            return result.IsSuccess ? onSuccess() : onFailure(result.Errors);
+        }
+
+        public static ActionResult CreateResponse<T>(
+            this Result<T> result,
+            Func<T, ActionResult> onSuccess, Func<List<IError>, ActionResult> onFailure)
+        {
+            return result.IsSuccess ? onSuccess(result.Value) : onFailure(result.Errors);
```

Comment:
Yes.

I don't see how moving it to BaseApiController would achieve the same effect, but I'm open to examples.

Reply:
I was thinking that instead of calling `CreateErrorResponse` from the base class as we do now, we call `CreateResponse` which can handle both success and failure scenarios.

Reply:
That we can do, nice catch. Implemented in the latest commit.